Develop

karvalolka opened the pull request.

karvalolka committed:
+ Request для store;
+ вывод количества перонажей во фракции и процентное соотношение;
+ роуты и контроллеры Store и Show;

karvalolka committed:
+ редактировать, удалить фракции

karvalolka committed:
+ создание рас (с выбором фракции);
+ создание классов (с выбором расы);
+ изменения в таблицах races и grades

kovdmit commented on app/Models/Fraction.php:
```diff
@@ -16,5 +16,10 @@ public function races()
     {
         return $this->hasMany(Race::class);
     }
+    public function char()
```
Лучше chars. Тут же будет коллекция

kovdmit commented on database/migrations/2024_09_11_170625_create_chars_table.php:
```diff
@@ -19,6 +19,8 @@ public function up(): void
             $table->unsignedSmallInteger('attack_power');
             $table->unsignedSmallInteger('def_power');
             $table->unsignedInteger('exp');
+            $table->unsignedBigInteger('fraction_id')->nullable();
```
Ты создаёшь новый PR , подразумевая, что у тебя продакшн уже работает. 

А если он работает, то у тебя не получится внести изменения в миграцию, чтобы она применялась без fresh. А освежение на проде удалит данные в бд, что недопустимо :(

Тебе следовало создать новую миграцию для изменения таблицы, а не редактировать старую.

p.s.: я не имею в виду, что твой подход недопустим, но для его применения нужны прям веские основания.

gotyefrid replied:
@karvalolka если миграция была уже применена однажды где-либо кроме твоей локалки - крайне не рекомендуется делать в ней правки. (если это не правки которые в принципе её чинять, если вдруг она какая то не рабочая оказалась и крашит всё)
@kovdmit +

karvalolka committed:
char->chars;
+ таблица race_fraction
- колонка name_class в таблице grade

karvalolka committed:
измененил связь расы с фракцией;
- лишние миграции;
отредактировал форму выбора фракции;
+ рабочее отображение принадлежности расы к фракции;

karvalolka committed:
+ рабочая ссылка на левой панеле (Главная);
+ ссылки на главной странице аналогичны тем, что находятся на левой панели с количеством;
+ ApiController;
+ JS для создания персонажа;
+ создание классов с привязкой к расам;
+ возможность создавать расу без привязке к ракции;
+ улучшения конфигурации;

karvalolka committed:
+ дополнительне разделы в сайдбаре;
+ routes, controllers, view (пока без img) для armor, weapon, attackRate, protectionArea и item (пока не рабочий);

karvalolka committed:
+ возможность создавать персонажа;
+ отображение email в карточке персонажа;

karvalolka committed:
+ возможность создавать персонажа;
+ отображение email в карточке персонажа;

karvalolka committed:
- колонка name в таблицу users;

karvalolka committed:
+ колонки image в таблицах armors и weapons;
+ валидация картинок в реквестах;
+ шаблоны для добавления картинок в blade;
+ отображение картинок;

karvalolka committed:
+ валидация типа урона;
+ создание оружия и брони без добавления картинок;
+ колонка name в таблице users;
+ роли
